app/src/App.tsx

```diff
@@ -39,6 +44,10 @@ type RootStackParamList = { [P in keyof typeof EXAMPLES]: undefined } & {
 };
 
 interface HomeScreenProps {
+  navigation: StackNavigationProp<RootStackParamList, 'Home'>;
+}
+
+interface HomeScreenNativeProps {
   navigation: NativeStackNavigationProp<RootStackParamList, 'Home'>;
 }
```

Comment:
Instead of having `function HomeScreen({ navigation }: HomeScreenProps | HomeScreenNativeProps)`, why don't we just use the union type here?

```suggestion
interface HomeScreenProps {
  navigation: StackNavigationProp<RootStackParamList, 'Home'> | NativeStackNavigationProp<RootStackParamList, 'Home'>;
}
```